pytorch_lightning/callbacks/lr_monitor.py

```diff
@@ -92,43 +95,62 @@ def on_train_start(self, trainer, *args, **kwargs):
 
         # Initialize for storing values
         self.lrs = {name: [] for name in names}
+        self.last_momentum_values = {name + "-momentum": None for name in names}
 
     def on_train_batch_start(self, trainer, *args, **kwargs):
         if not self._should_log(trainer):
             return
 
         if self.logging_interval != 'epoch':
             interval = 'step' if self.logging_interval is None else 'any'
-            latest_stat = self._extract_lr(trainer, interval)
+            latest_stat = self._extract_stats(trainer, interval)
 
             if trainer.logger is not None and latest_stat:
                 trainer.logger.log_metrics(latest_stat, step=trainer.global_step)
 
     def on_train_epoch_start(self, trainer, *args, **kwargs):
         if self.logging_interval != 'step':
             interval = 'epoch' if self.logging_interval is None else 'any'
-            latest_stat = self._extract_lr(trainer, interval)
+            latest_stat = self._extract_stats(trainer, interval)
 
             if trainer.logger is not None and latest_stat:
                 trainer.logger.log_metrics(latest_stat, step=trainer.current_epoch)
 
-    def _extract_lr(self, trainer, interval: str) -> Dict[str, float]:
+    def _extract_stats(self, trainer, interval: str) -> Dict[str, float]:
         latest_stat = {}
 
         for name, scheduler in zip(self.lr_sch_names, trainer.lr_schedulers):
             if scheduler['interval'] == interval or interval == 'any':
                 param_groups = scheduler['scheduler'].optimizer.param_groups
                 if len(param_groups) != 1:
                     for i, pg in enumerate(param_groups):
-                        lr, key = pg['lr'], f'{name}/pg{i + 1}'
-                        self.lrs[key].append(lr)
-                        latest_stat[key] = lr
+                        lr = self._extract_lr(param_group=pg, name=f'{name}/pg{i + 1}')
+                        latest_stat.update(lr)
+                        momentum = self._extract_momentum(param_group=pg, name=f'{name}-momentum/pg{i + 1}')
+                        latest_stat.update(momentum)
+
                 else:
```

Comment:
Is this added to follow existing conventions in the code base? Otherwise seems a bit redundant?
(removing the idx suffix for single param groups)

Reply:
yes, i'm mirroring the existing convention for learning rates